src/data/real/ereal.lean

```diff
@@ -121,6 +125,38 @@ protected def mul : ereal → ereal → ereal
 
 instance : has_mul ereal := ⟨ereal.mul⟩
 
+/-- Induct on two ereals by performing case splits on the sign of one whenever the other is
+infinite. -/
+@[elab_as_eliminator]
+lemma induction₂ {P : ereal → ereal → Prop}
```

Comment:
The cases don't overlap, right? So we could turn that into a recursor.
```suggestion
lemma rec₂ {P : ereal → ereal → Sort*}
```

Reply:
I could do this (and use it to define `mul`), but then I'd need to insert some `eq.rec`s as well. The resulting `mul` would be definitionally harder to unfold, and also harder to read; so I'd prefer not to bother.

Reply:
Yeah that's a fair point.